### ISSUE TITLE
SD3 LoRA training time

### ISSUE THREAD

**Issue by primecai** [bghira/SimpleTuner#613]
Hi,

Thanks for this nice repo!
I have been trying to train a LoRA on SD3 using the multi-GPU setting. I am on 4 48GPUs (A6000), and my dataset is 1000 1024x1024images --- I set up the batch size as 10 and do no gradient accumulation. Each iteration takes 40-50 seconds to complete.
It seems this training speed is dramatically slower comparing with many of the other logs I find in issues of this repo. Is it a normal phenomenon, or potentially I have something wrong setups?

Best regards and thanks in advance.

**Comment by bghira** [comment]
you need --vae_cache_preprocess

**Comment by primecai** [comment]
I added `--vae_cache_preprocess` in the bash script, but it seems the training time is still pretty much the same (it may have been reduced by 15%). Is ~40 seconds per iteration somewhat expected? I saw logs in other issues, such as [this one](https://github.com/bghira/SimpleTuner/issues/464), to be 1–2 seconds per iteration.

**Comment by bghira** [comment]
can you try with just one GPU?

**Comment by bghira** [comment]
for what it's worth, 40 seconds is a lot more than i'd expect. especially for a LoRA on a 2B model. it should be more like 3-5 seconds per second at worst, and 10 seconds per step when training from S3 backend
